lib/battle_boats/random_fleet_deployer.rb

```diff
@@ -0,0 +1,30 @@
+require_relative "board"
+require_relative "coordinate"
+
+module BattleBoats
+  class RandomFleetDeployer
+    def initialize(board: BattleBoats::Board.new(state: :enemy))
```

Comment:
can you clarify what state enemy means for the board class here?

Reply:
In this case it's a control parameter code smell. The `state: enemy` flag will return a `Board` instance that is made up of `Cell`s in an "enemy" state. 

This is an artifact of my attempt at forcing these objects into the state pattern.

Reply:
This is actually incorrect code. line 6 should be:

```
def initialize(board: BattleBoats::Board.for(state: :enemy))
```

Not that that makes it any better. `BattleBoats::Board.for` is a factory which returns a new `BattleBoats::Board` with `Cell`s in the given state. 

But, as we talked about `enemy` and `ally` aren't really states, their only purpose is to determine display logic, and there are cleaner alternatives to this.